### DISCUSSION TITLE
Using 48kHz files

### DISCUSSION THREAD

**Discussion by Pythagorstudio** [swesterfeld/audiowmark#52] [Q&A]
Hi
From Developer papers I understood that audiowmark uses 44.1 conversion. I wonder if it is possible to implement watermarking of 48 kHz files without conversion. 

Thanks

**Comment by swesterfeld** [comment]
Hi!

I don't really see a compelling reason why you would want to change the default behaviour, unless you want to squeeze out a bit of performance. Zita-Resampler is fast, but it does have a performance impact since the actual watermarking is also quite fast.

From a quality point of view, there should be no problem, because the resampling steps only affect the watermark signal, but the original remains untouched. So for a 48 kHz file we do these steps:

1. downsample original to 44.1
2. generate watermark at 44.1
3. upsample watermark to 48 kHz
4. add original 48 kHz signal to the watermark signal from step 3

Since 3 should be inaudible and the original signal is added (without any conversion), there should be no quality issue. That said, you can change `mark_sample_rate` in `wmcommon.h` to 48000. Only keep in mind that the audiowmark version generating the
watermark and detecting the watermark need to use the same value here.

Keep in mind that changing one parameter here affects how the watermark sounds. For the change from 44.1 to 48 kHz this should be minimal, so you should not run into quality issues. If you were to perform extreme changes here (like changing it to 96 kHz or 22050) you may need to adjust other parameters to maintain good subjective quality.
